msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -166,21 +163,39 @@ public Integer getNumberOfDataPoints() {
 
   @Override
   public MsSpectrumType getSpectrumType() {
-    MsSpectrumType spectrumType = SpectrumTypeDetectionAlgorithm.detectSpectrumType(mzValues,
-        intensityValues, getMzBinaryDataInfo().getArrayLength());
-    return spectrumType;
+    if (getCVValue(MzMLCV.cvCentroidSpectrum) != null)
+      return MsSpectrumType.CENTROIDED;
+
+    if (getCVValue(MzMLCV.cvProfileSpectrum) != null)
+      return MsSpectrumType.PROFILE;
+
+    return null;
   }
 
   @Override
   public Float getTIC() {
-    Float tic = MsSpectrumUtil.getTIC(intensityValues, getMzBinaryDataInfo().getArrayLength());
+    Float tic = null;
+    try {
+      tic = Float.valueOf(getCVValue(MzMLCV.cvTIC));
+    } catch (Exception e) {
```

Comment:
It is not a good practice to rely on exceptions for common situations like if the TIC CV param is missing - exceptions should be thrown in truly exceptional cases. In this case, you should first check if getCVValue returns null, and then try to parse it (in a try/catch block).